### ISSUE TITLE
Program crashes with exit code -1073741819 (0xC0000005)

### ISSUE THREAD

**Issue by benjibst** [glfw/glfw#2503]
I tried to set up GLFW, built from source, with glad
Program crashes with exit code -1073741819 (0xC0000005) on glfwCreateWindow call. To debug, i rebuilt GLFW with Debug config and i found the function that makes the program crash, it seems as though address 0 is called as a function. I put the exact place where it fails and the disassembly here.
![image](https://github.com/glfw/glfw/assets/76067254/913f9c08-c614-41c1-9f78-35cc14178f7d)
![image](https://github.com/glfw/glfw/assets/76067254/1a7e2a1b-2c2f-44f0-9951-12fa556b0ffc)

for anyone wanting to reproduce you can find the code that lead to this error here https://github.com/benjibst/fractals/commit/8eb5eb6389081698d9b6ca5593ee568c9a74e08f

**Comment by elmindreda** [comment]
That shouldn't be happening. Are you using an Intel GPU?

**Comment by benjibst** [comment]
No, this is happening only on my windows 10 pc (on my debian laptop it works fine both on integrated and discrete gpu). Pc configuration is AMD ryzen 5 1600x, gtx 1070, 16gb ddr4,

**Comment by dougbinks** [comment]
Could you try running the tests or examples from the GLFW source? This would help narrow down the problem.

The easiest way to do this is just use cmake with default options to generate Visual Studio C++ build files and run them from there.

**Comment by dougbinks** [comment]
I thought I would add some further information.

The symbol `wglMakeCurrent` in GLFW is a macro `_glfw.wgl.MakeCurrent` which is loaded from `opengl32.dll` by 
https://github.com/glfw/glfw/blob/master/src/wgl_context.c#L433-L434

If you put a breakpoint after these are loaded you can check if the function pointer is non zero. This shouldn't happen unless `opengl32.dll` has been replaced or loaded from a non system directory.

`opengl32.dll` is normally a system library (i.e. not from the GPU driver), but it is possible (and useful sometimes) to use another version, for example to use software rendering. But in your case this library should be loaded from the system directory. If you're running from Visual Studio you can see output like:

`(Win32): Loaded 'C:\Windows\System32\opengl32.dll'. `

There are some game overlay applications which hook functions in the library, and these can cause crashes but I've not seen this one.

Updating drivers for both the AMD integrated GPU and NVIDIA GPU might help, though again I've not see this particular issue.

**Comment by dougbinks** [comment]
I see you are using CMake for your project build configuration, in which case you may want to look at:

https://github.com/juliettef/GLFW-CMake-starter

This pulls GLFW source from Github as a subproject, and uses the GLFW CMake file for building.

**Comment by dougbinks** [comment]
If you are still having the same problem then one possibility is that in between `glfwInit()` and the window creation call something is overwriting the gflw data structure. If so you can add a memory address change breakpoint:

1. Put a breakpoint in `_glfwInitWGL(void)` after `_glfw.wgl.MakeCurrent ` is initialized.
2. hover the mouse over `_glfw.wgl.MakeCurrent` until you see the variable and it's value appear, then right click on the popup window and select `Break when value changes`. You should also be able to see that the value of this variable is non-zero. If it is zero then `OpenGL32.dll` has been loaded from the wrong place - check the debug output window to see where it was loaded from.
3. Continue debugging, this should break when the value is overwritten.

See below:
![image](https://github.com/glfw/glfw/assets/872946/f0ad3d78-6454-4c3a-8fb4-7028e54d4374)

Note that I haven't updated to the latest code, so your line numbers will be different. Additionally the location of `_glfw.wgl.MakeCurrent` depends on the module address which is always different.

**Comment by benjibst** [comment]
Very weird... I built the project in Visual Studio to follow your guide on debugging, and the program works now. Maybe the issue is because i forgot to link some library that is linked automatically in Visual Studio but not in Clion?

**Comment by dougbinks** [comment]
This error shouldn't occur because you didn't link a library, since the functions are loaded directly by GFLW.

You might have a buffer overflow (or similar) which doesn't effect the VS build the same way as the Clion one. If you experience other issues you can also check out the [AppVerifier tool from Microsoft](https://learn.microsoft.com/en-us/windows-hardware/drivers/devtest/application-verifier) which can help debug memory corruptions.

**Comment by benjibst** [comment]
Tried to do the same type of debugging in CLion too, i set a "watchpoint" on the variable to see when it changes. It is set correctly, but the program doesnt hit the breakpoint before the call to zero is executed.

**Comment by dougbinks** [comment]
Does `_glfw.wgl.MakeCurrent` get set to a non zero value in the `_glfwInitWGL(void)` function? If so then I think you should be able to use AppVerifier (as above) with CLion. It will trigger a debug break when an issue is detected.

**Comment by benjibst** [comment]
Yes, it gets set to a non zero value. Ill try the AppVerifier sometime

**Comment by dougbinks** [comment]
Another thought is that you might somehow have more than one copy of GLFW loaded, for example if you link to a static version of GLFW in the executable and in a DLL. If you initialize in one module and create the window in another this could then cause an issue.

To ensure you have only one version of GLFW in a multi-module program you can link to the DLL version.

**Comment by benjibst** [comment]
Ill verify that too, thx

**Comment by dougbinks** [comment]
Thinking a bit more about it, multiple GLFWs are unlikely to be the problem, as `glfwCreateWindow` calls ` _GLFW_REQUIRE_INIT_OR_RETURN(NULL);` and so would exit returning `NULL` rather than dereferencing a `NULL` pointer later on.

**Comment by dougbinks** [comment]
> I built the examples and they run just fine, maybe theres just a problem with my code

Since this does not occur in GLFW examples and appears to be an old support issue I will close this, however if you still have the issue please reopen, create a new issue or ask for help on https://discourse.glfw.org/ if needed.
